File: Kingdom-Fall-UI-Enabled/src/main/java/objets/Divers.java
package objets;

public abstract class Divers extends Objet {

    private final String description;

    public Divers (String nom, String description, double dropRate)
    {
        super (Type_Objet.DIVERS, nom, dropRate);
        this.description = description;
    }

    public Divers (Divers autre)
    {
        super (autre);
        this.description = autre.description;
    }

    @Override
    public String getDescription() {
        return String.format("%s%n%s%nChances de drop : %.2f%n", nom, description, dropRate);
    }

    public void utiliser () {}
}

File: KF-Remake/Kingdom-Fall/source/objets/Divers.cs
namespace objets;

public abstract class Divers : Objet
{
    private string description;

    public Divers(string nom, string description, double dropRate)
        : base(TypeObjet.Divers, nom, dropRate)
    {
        this.description = description;
    }

    public Divers(Divers autre) : base(autre)
    {
        this.description = autre.description;
    }

    public override string GetDescription()
    {
        return $"{nom}\n{description}\nChances de drop : {dropRateString}";
    }

    public virtual void utiliser() {}
}
